pysnmp/proto/secmod/eso/priv/des3.py

```diff
@@ -51,17 +52,19 @@ def hashPassphrase(self, authProtocol, privKey):
                 'Unknown auth protocol %s' % (authProtocol,)
             )
 
-    # 2.1
+    #key localization as per https://tools.ietf.org/html/draft-reeder-snmpv3-usm-3desede-00
     def localizeKey(self, authProtocol, privKey, snmpEngineID):
         if authProtocol == hmacmd5.HmacMd5.serviceID:
             localPrivKey = localkey.localizeKeyMD5(privKey, snmpEngineID)
-            for count in range(1, int(ceil(self.keySize * 1.0 / len(localPrivKey)))):
-                # noinspection PyDeprecation,PyCallingNonCallable
-                localPrivKey += localkey.localizeKeyMD5(localPrivKey, snmpEngineID)
+            #now extend this key if too short by repeating steps that includes the hashPassphrase step
+            while (len(localPrivKey) < self.keySize):
+                newKey = hashPassphraseMD5(localPrivKey)
+                localPrivKey = localPrivKey + localizeKeyMD5(newKey, snmpEngineID)
         elif authProtocol == hmacsha.HmacSha.serviceID:
             localPrivKey = localkey.localizeKeySHA(privKey, snmpEngineID)
-            for count in range(1, int(ceil(self.keySize * 1.0 / len(localPrivKey)))):
-                localPrivKey += localkey.localizeKeySHA(localPrivKey, snmpEngineID)
+            while (len(localPrivKey) < self.keySize):
+                newKey = hashPassphraseSHA(localPrivKey)
+                localPrivKey = localPrivKey + localizeKeySHA(newKey, snmpEngineID)
```

Comment:
Would not:

    localPrivKey += localizeKeySHA(newKey, snmpEngineID)

syntax be a little more concise?